### PR TITLE
Added XTYPES switch to RTPS Discovery configuration

### PR DESCRIPTION
**Pull request by kuznetsovmoci**
Added XTYPES switch to RTPS Discovery configuration. Made Type Information addition to the parameter list conditional.

### CODE REVIEW

File: dds/DCPS/RTPS/Sedp.cpp [hunk 1]
```diff
@@ -497,6 +495,8 @@ Sedp::init(const RepoId& guid,
   }
 #endif
 
+  max_type_lookup_service_reply_period_ = disco.config()->max_type_lookup_service_reply_period();
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Sedp.cpp:498]: Why is this line changing?

**Review by mitza-oci — CHANGES_REQUESTED** [review]
Should this also control the `from_param_list` direction?  If we don't change that code, the local participant can receive an endpoint announcement from a peer that includes type information and process it -- doesn't seem like what we want.

File: dds/DCPS/RTPS/Sedp.cpp (hunk 1 above)
Comment (kuznetsovmoci) [reply dds/DCPS/RTPS/Sedp.cpp:498]: I moved it from above, where it was in the middle of writers /  readers setup.

**Review by mitza-oci — APPROVED** [review]
.

**Pull request merged**